plotly/plotly/plotly.py

```diff
@@ -1342,6 +1345,114 @@ def get_grid(grid_url, raw=False):
+        :param (str) filename: the name of the dashboard to be saved in
+            your Plotly account. Will overwrite a dashboard of the same
+            name if it already exists in your files.
+        :param (str) sharing: can be set to either 'public', 'private'
+            or 'secret'. If 'public', your dashboard will be viewable by
+            all other users. If 'secret', only you can see your dashboard.
+            If 'secret', the url will be returned with a sharekey appended
+            to the url. Anyone with the url may view the dashboard.
+        :param (bool) auto_open: automatically opens the dashboard in the
+            browser.
+        """
+        if sharing == 'public':
+            world_readable = True
+        elif sharing == 'private':
+            world_readable = False
+        elif sharing == 'secret':
+            world_readable = False
+
+        data = {
+            'content': json.dumps(dashboard),
+            'filename': filename,
+            'world_readable': world_readable
+        }
+
+        # lookup if pre-existing filename already exists
+        try:
+            v2.files.lookup(filename)
+            matching_dashboard = cls._get_dashboard_json(
+                filename, False
+            )
+
+            if matching_dashboard['filetype'] == 'dashboard':
+                old_fid = matching_dashboard['fid']
+                res = v2.dashboards.update(old_fid, data)
+
+        except exceptions.PlotlyRequestError:
+            res = v2.dashboards.create(data)
+        res.raise_for_status()
+
+        url = res.json()['web_url']
+
+        if auto_open:
+            webbrowser.open_new(res.json()['web_url'])
+
+        if sharing == 'secret':
+            url = add_share_key_to_url(url)
+
+        return url
+
+    @classmethod
+    def _get_all_dashboards(cls):
+        dashboards = []
+        res = v2.dashboards.list().json()
+
+        for dashboard in res['results']:
+            if not dashboard['deleted']:
+                dashboards.append(dashboard)
+        while res['next']:
+            res = v2.utils.request('get', res['next']).json()
+
+            for dashboard in res['results']:
+                if not dashboard['deleted']:
+                    dashboards.append(dashboard)
+        return dashboards
+
+    @classmethod
+    def _get_dashboard_json(cls, dashboard_name, only_content=True):
+        dashboards = cls._get_all_dashboards()
+        for index, dboard in enumerate(dashboards):
+            if dboard['filename'] == dashboard_name:
+                break
+
+        dashboard = v2.utils.request(
+            'get', dashboards[index]['api_urls']['dashboards']
+        ).json()
+        if only_content:
+            dashboard_json = json.loads(dashboard['content'])
+            return dashboard_json
+        else:
+            return dashboard
+
+    @classmethod
+    def get_dashboard(cls, dashboard_name):
+        """Returns a Dashboard object from a dashboard name."""
```

Comment:
`help(py.meta_ops.upload)` also results in the same message.

I think it's fine since `help(py.dashboard_ops)` will display the doc strings for all the other methods in the class